client/my-sites/plugins/controller-logged-out.js

```diff
@@ -36,11 +42,50 @@ const prefetchPopularPlugins = ( queryClient, options ) =>
 const prefetchFeaturedPlugins = ( queryClient ) =>
 	prefetchPluginsData( queryClient, getWPCOMFeaturedPluginsQueryParams() );
 
-const prefetchProductList = ( store ) => {
-	const productsList = getProductsList( store.getState() );
-	if ( Object.values( productsList ).length === 0 ) {
-		return requestProductsList( { type: 'all' } )( store.dispatch );
+const prefetchProductList = ( queryClient, store ) => {
+	const query = { type: 'all' };
```

Comment:
Little nit: it's possible to write `const type = 'all'` and then `req.get( '/products', { type } )` and `receive( productsList, type )`, I think it's a bit nicer.